### PR TITLE
[discover] fix too many queries sent

### PR DESCRIPTION
**Pull request by kavilla**
### Description

Fixes the issue where too many queries were being fired. A number of things were being called because the dependency array was triggering updates and re-calls. Forcing a re-render of elements that had the possibility of updating the query manager.

Includes some tests.

### Issues Resolved

n/a

## Testing the changes

locally and tests

## Changelog

- fix: prevent too many queries sent from dataset selector

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by abbyhu2000 — APPROVED** [review]

File: src/plugins/data/public/ui/dataset_selector/index.tsx [hunk 1]
```diff
@@ -17,25 +17,21 @@ interface ConnectedDatasetSelectorProps {
...
-  const initialDataset = queryString.getQuery().dataset || queryString.getDefaultQuery().dataset;
-  const [selectedDataset, setSelectedDataset] = useState<Dataset | undefined>(initialDataset);
-
-  useEffect(() => {
-    const subscription = queryString.getUpdates$().subscribe((query) => {
-      setSelectedDataset(query.dataset);
-    });
-
-    return () => subscription.unsubscribe();
-  }, [queryString]);
```
Comment (sejli) [inline src/plugins/data/public/ui/dataset_selector/index.tsx:29]: I remember being confused why this is here. Was there a reason why we needed two places to set the selected dataset?

**Review by sejli — APPROVED** [review]
Amazing

File: src/plugins/data/public/ui/dataset_selector/index.tsx (hunk 1 above)
Comment (kavilla) [reply src/plugins/data/public/ui/dataset_selector/index.tsx:29]: not positive. probably was just code that was intended to be cleaned up i think i at one point we would set the query at the configurator level instead of using the function that we passed in to set it and then when we got time to wire up and use the passed in function it caused some issue with re-rendering.

File: src/plugins/data/public/query/query_string/query_string_manager.ts [hunk 2]
```diff
@@ -68,6 +68,7 @@ export class QueryStringManager {
     const query = {
       query: defaultQuery,
       language: defaultLanguageId,
+      dataset: undefined,
```
Comment (abbyhu2000) [inline src/plugins/data/public/query/query_string/query_string_manager.ts:71]: why is this necessary? i see query is used to construct `newQuery` and dataset is set to be defaultDataset in line 81

Comment (kavilla) [reply src/plugins/data/public/query/query_string/query_string_manager.ts:71]: my linter was complaining because i thought the return type was just query language

**Review by joshuali925 — APPROVED** [review]
nice, lgtm

**Review by sejli — APPROVED** [review]

**Review by abbyhu2000 — APPROVED** [review]

**Comment by kavilla** [comment]
failures unrelated and i think another PR is fixing it.
